tests/store/test_cli.py

```diff
@@ -26,6 +27,46 @@ def test_file_info_to_json():
     }]
 
 
+def test_download_from_uri():
+    class TestArtifactRepo:
+        def __init__(self, scheme):
+            self.scheme = scheme
+
+        def download_artifacts(self, artifact_path, dst_path=None):
+            return (self.scheme, artifact_path)
+
+    def test_get_artifact_repository(artifact_uri, **kwargs):
+        return TestArtifactRepo(artifact_uri)
+
+    pairs = [
+        ("path", ("", "path")),
+        ("path/", ("path", "")),
+        ("/path", ("/", "path")),
+        ("/path/", ("/path", "")),
+        ("path/to/dir", ("path/to", "dir")),
+        ("file:///", ("file:///", "")),
+        ("file:///path", ("file:///", "path")),
+        ("file:///path/", ("file:///path", "")),
+        ("file:///path/to/dir", ("file:///path/to", "dir")),
+        ("s3://", ("s3:", "")),
+        ("s3://path", ("s3://path", "")),  # path is netloc in this case
+        ("s3://path/", ("s3://path/", "")),
+        ("s3://path/to/", ("s3://path/to", "")),
+        ("s3://path/to", ("s3://path/", "to")),
+        ("s3://path/to/dir", ("s3://path/to", "dir")),
+    ]
+    orig = mlflow.store.artifact_repository_registry._artifact_repository_registry. \
```

Comment:
It's probably a better idea to use [mock](https://pypi.org/project/mock/) here and mock the [get_artifact_repository](https://github.com/mlflow/mlflow/blob/8582b5b83aa1896ac7af6423572a0080425db006/mlflow/store/artifact_repository_registry.py#L89) method, rather than its implementation. Using mock will also ensure that an unexpected failure in this individual test case does not cause subsequent failures for all test cases depending on `get_artifact_repository`.

Reply:
This should not affect other tests, it's in a try/finally,

I tried to use mock but it was too difficult to get it to do what I want. If you know how can this be done with mock easily I'll be happy to change it ;)